### PR TITLE
Pact 4.3.1 upgrade and tests

### PR DESCRIPTION
**Pull request by jmcardon**

### CODE REVIEW

File: src/Chainweb/Version.hs [hunk 1]
```diff
@@ -943,6 +944,21 @@ chainweb214Pact aoa v h = case aoa of
...
     go f _ = f 5
 
+chainweb215Pact
+    :: AtOrAfter
+    -> ChainwebVersion
+    -> BlockHeight
+    -> Bool
+chainweb215Pact aoa v h = case aoa of
+    At -> go (==) v h
+    After -> go (flip (>)) v h
```
Comment (emilypi) [inline src/Chainweb/Version.hs:954]:
```suggestion
    After -> go (<) v h
```

File: src/Chainweb/Version.hs [hunk 2]
```diff
@@ -935,14 +936,29 @@ chainweb214Pact
     -> Bool
 chainweb214Pact aoa v h = case aoa of
     At -> go (==) v h
-    After -> go (flip (>)) v h
+    After -> go (<=) v h
```
Comment (larskuhtz) [inline src/Chainweb/Version.hs:939]: This changes the meaning: `flip (>) /== (<=)`. Is that indented?

File: src/Chainweb/Version.hs [hunk 3]
```diff
@@ -935,14 +936,29 @@ chainweb214Pact
...
     go f _ = f 5
 
+chainweb215Pact
+    :: AtOrAfter
+    -> ChainwebVersion
+    -> BlockHeight
+    -> Bool
+chainweb215Pact aoa v h = case aoa of
+    At -> go (==) v h
+    After -> go (<=) v h
```
Comment (larskuhtz) [inline src/Chainweb/Version.hs:954]: See above regarding `<=` vs `<`

Comment (larskuhtz) [inline src/Chainweb/Version.hs:954]: See above regarding `<=` vs `<`

File: src/Chainweb/Version.hs [hunk 4]
```diff
@@ -886,7 +887,7 @@ data AtOrAfter = At | After deriving (Eq,Show)
 pact4coin3Upgrade :: AtOrAfter -> ChainwebVersion -> BlockHeight -> Bool
 pact4coin3Upgrade aoa v h = case aoa of
     At -> go (==) v h
-    After -> go (flip (>)) v h
+    After -> go (<=) v h
```
Comment (larskuhtz) [inline src/Chainweb/Version.hs:890]: This changes the meaning: flip (>) ≠ (<=). Is that indented?

**Review by larskuhtz — APPROVED** [review]

**Pull request merged**